python/tank/context.py

```diff
@@ -806,7 +808,9 @@ def _fields_from_shotgun(self, template, entities, validate):
                 
                 # check the context cache 
                 cache_key = (entity["type"], entity["id"], key.shotgun_field_name)
-                if cache_key in self._entity_fields_cache:
+                # MOFA - test for actual cache value, can be None here ...
+                # if cache_key in self._entity_fields_cache:
+                if cache_key in self._entity_fields_cache and self._entity_fields_cache[cache_key] is not None:
```

Comment:
I think this can be shorted down to `if self._entity_fields_cache.get(cache_key):` ? 
It would be great with a comment explaining the case when we are expecting null values inside the `_entity_fields_cache` dict.